src/engine/Operation.cpp

```diff
@@ -69,17 +69,38 @@ void Operation::recursivelySetTimeConstraint(
   });
 }
 
+// _____________________________________________________________________________
+void Operation::updateRuntimeStats(bool applyToFilter, uint64_t numRows,
+                                   uint64_t numCols,
+                                   std::chrono::milliseconds duration) const {
+  auto& rti = applyToFilter || !externalFilterApplied_
```

Comment:
You really need to document `applyToFilter`and `externalFilterApplied_` at least here, I don't quite understand it.